UH-StudentServices/student_guide
# Feature/hub 217 degree programmes api

# Testing

- Use a REST testing tool (such as the one in PhpStorm)

- GET `https://local.guide.student.helsinki.fi/api/v1/degree-programme?_format=json`

- Verify that all the degree programmes are returned (code and name translations)

- Verify that subsequent calls are returned from cache

- Go and edit one of the degree programme terms, it is easiest to modify the name of the first one, so that it is easy to see if the change comes through in the API call

- IMPORTANT: After editing (always), remember to run `drush pqw` (p-queue-work, "Claim a chunk of items from the queue and process them."), this will invalidate the response in Varnish

- GET `https://local.guide.student.helsinki.fi/api/v1/degree-programme?_format=json`

- Verify that the change is present in the response
